ui/layout.js

```diff
@@ -16,22 +16,106 @@
  * along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+const { Clutter, GObject, Shell, St } = imports.gi;
+
+const ExtensionUtils = imports.misc.extensionUtils;
+const DesktopExtension = ExtensionUtils.getCurrentExtension();
+
+const AppDisplay = imports.ui.appDisplay;
+const AppDisplayOverrides = DesktopExtension.imports.ui.appDisplay;
+const LayoutManager = imports.ui.layout;
 const Main = imports.ui.main;
 
+const EOS_INACTIVE_GRID_OPACITY = 96;
+
 let startupPreparedId = 0;
 
+var OverviewClone = GObject.registerClass(
+class OverviewClone extends St.BoxLayout {
+    _init() {
+        super._init({
+            name: 'overview',
+            opacity: EOS_INACTIVE_GRID_OPACITY,
+            vertical: true,
+        });
+
+        Shell.util_set_hidden_from_pick(this, true);
+
+        this.add_constraint(new LayoutManager.MonitorConstraint({ primary: true }));
+
+        // Add a clone of the panel to the overview so spacing and such is
+        // automatic
+        const panelGhost = new St.Bin({
+            child: new Clutter.Clone({ source: Main.panel }),
+            reactive: false,
+            opacity: 0,
+        });
+        this.add_child(panelGhost);
```

Comment:
I was planning to hide the `panelGhost` when moving the panel to the bottom in the panel extension but with this it will make it harder. I guess we can keep the ghost and its clone around, but something to check at a later point.

Otherwise lgtm, merging.